public/app/features/query/state/DashboardQueryRunner/AnnotationsWorker.ts

```diff
@@ -50,7 +50,7 @@ export class AnnotationsWorker implements DashboardQueryRunnerWorker {
       let datasourceObservable;
 
       if (config.publicDashboardAccessToken) {
-        const pubdashDatasource = new PublicDashboardDataSource(PUBLIC_DATASOURCE);
+        const pubdashDatasource = new PublicAnnotationsDataSource();
```

Comment:
Is this still necessary?  won't the normal path work ok?  (and if not, would be an error anyway)

Reply:
So, we still need to have a class that inherits from DataSourceApi and has a query method implemented for the annotations flow.
That's because the normal flow takes a specific datasource (Grafana datasource) and executes the query method to get the annotations.

We could try getting the Grafana datasource and add an `if` in the query method to call the public dashboard endpoint or continue with the normal flow, but I don't want to add too many modifications to this PR.
Thoughts? @torkelo

Reply:
yea, we can look at that in a separate PR if it makes sense.